keps/sig-node/4800-cpumanager-split-uncorecache/README.md

```diff
@@ -0,0 +1,689 @@
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+To enable this feature requires adding to the Kubelet yaml file for the CPUManager feature gate and add the policy option for uncore cache alignment
+
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+For `CPUManager` it is a requirement going from `none` to `static` policy cannot be done dynamically because of the `cpu_manager_state file`. The node needs to be drained and the policy checkpoint file (`cpu_manager_state`) need to be removed before restarting Kubelet. This feature specifically relies on the `static` policy being enabled.
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `CPUManagerAlphaPolicyOptions`
+  - Components depending on the feature gate: `kubelet`
+- [x] Other
+  - Describe the mechanism: Change the `kubelet` configuration to set a `CPUManager` policy of static then setting the policy option of `align-cpus-by-uncorecache`
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No.
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? Yes, a `kubelet` restart is required for changes to take place.
+
+###### Does enabling the feature change any default behavior?
+
+No, to enable this feature, it must be explicitly set in the `CPUManager` static policy and the policy option `align-cpus-by-uncorecache` must be set.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes this feature can be disabled it will just require a restart of `kubelet`. The Kubelet configuration will need to be set with the static policy option and align-cpus-by-uncorecache flag removed.
+
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Feature will be enabled. Proper drain of node and restart of kubelet required. Feature is not intended to be enabled/disabled dynamically, similar to static policy.
+
+###### Are there any tests for feature enablement/disablement?
+
+Option is not enabled dynamically. To enable/disable option, cpu_manager_state must be removed and kubelet must be restarted.
+Unit tests will be implemented to test if the feature is enabled/disabled.
+E2e node serial suite can be use to test the enablement/disablement of the feature since it allows the kubelet to be restarted.
+
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+Kubelet restarts are not expected to impact existing CPU assignments to already running workloads
+
+
+###### What specific metrics should inform a rollback?
+
+N/A
```

Comment:
the reason being the alignment is best-effort, so it cannot cause additional failures

Reply:
I'd suggest adding increased  pod startup latency as an information for a rollback, at minimum.

Reply:
yes, that would make sense

Reply:
I'm not very clear on this, I get the pod startup latency but I'm not understanding how this relates to a rollback, since a rollback would require draining the node

Reply:
Cluster administrator do monitor their cluster after they upgraded, and they need a way to learn if the new features they deployed are working or not; if not, they may want to rollback their cluster (or, hopefully, their test nodes on green/blue deployments) to the previous state. Which brings the question: how should cluster admins learn if the feature is working, if the cluster is working, if the affected nodes are working, thus they may need a rollback? Hence the PRR question.

Example: pod admission time (startup time) "too high" -> node misbehaving -> rollback to safe state to keep serving production traffic

Reply:
added